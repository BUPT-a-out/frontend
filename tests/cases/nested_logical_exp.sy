int main() {
    int a = 1, b = 2;

    if (a < b || a > 0 && b > 0)
        a = a + 1;

    if (a < b || a > 0 && b > 0)
        a = a + 1;
    else
        a = a - 1;

    while (a < b || a > 0 && b > 0) {
        a = a + 1;
    }

    return 0;
}
