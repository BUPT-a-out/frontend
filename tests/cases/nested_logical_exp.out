Parsing completed successfully.

--- Abstract Syntax Tree ---
+-- ROOT
|   +-- FUNC_DEF: main (sym: main, id: 26)
|   |   +-- LIST: FParams
|   |   +-- LIST: Block
|   |   |   +-- IF_ELSE_STMT
|   |   |   |   +-- CONST (int value: 1)
|   |   |   |   +-- LIST: If-2
|   |   |   |   |   +-- IF_ELSE_STMT
|   |   |   |   |   |   +-- CONST (int value: 1)
|   |   |   |   |   |   +-- LIST: If-2
|   |   |   |   |   |   +-- LIST: If-3
|   |   |   |   +-- LIST: If-3
|   |   |   |   |   +-- WHILE_STMT
|   |   |   |   |   |   +-- CONST (int value: 0)
|   |   |   |   |   |   +-- LIST: While-2
|   |   |   |   |   |   |   +-- IF_ELSE_STMT
|   |   |   |   |   |   |   |   +-- CONST (int value: 1)
|   |   |   |   |   |   |   |   +-- LIST: If-2
|   |   |   |   |   |   |   |   +-- LIST: If-3
|   |   |   +-- RETURN_STMT: main (sym: main, id: 26)
|   |   |   |   +-- CONST (int value: 0)

--- Permanent Symbol Table ---
ID    Name                 Type            Data Type  Function             Shape     
-------------------------------------------------------------------------------------
0     getint               function        int        N/A                  N/A       
1     getch                function        int        N/A                  N/A       
2     getfloat             function        float      N/A                  N/A       
3     getarray             function        int        N/A                  N/A       
4     array                array           int        getarray             0
5     getfarray            function        int        N/A                  N/A       
6     array                array           float      getfarray            0
7     putint               function        void       N/A                  N/A       
8     value                var             int        putint               N/A       
9     putch                function        void       N/A                  N/A       
10    value                var             int        putch                N/A       
11    putfloat             function        void       N/A                  N/A       
12    value                var             float      putfloat             N/A       
13    putarray             function        void       N/A                  N/A       
14    len                  var             int        putarray             N/A       
15    array                array           int        putarray             0
16    putfarray            function        void       N/A                  N/A       
17    len                  var             int        putfarray            N/A       
18    array                array           float      putfarray            0
19    putf                 function        void       N/A                  N/A       
20    format_string        var             char       putf                 N/A       
21    value                var             int        putf                 N/A       
22    starttime            function        void       N/A                  N/A       
23    line                 var             int        starttime            N/A       
24    stoptime             function        void       N/A                  N/A       
25    line                 var             int        stoptime             N/A       
26    main                 function        int        N/A                  N/A       
-------------------------------------------------------------------------------------

--- Generated IR ---
; ModuleID = 'main'

define i32 @main() {
main.entry:
  br i1 1, label %if.0.then, label %if.0.else
if.0.then:
  br i1 1, label %if.1.then, label %if.1.else
if.1.then:
  br label %if.1.merge
if.1.else:
  br label %if.1.merge
if.1.merge:
  br label %if.0.merge
if.0.else:
  br label %while.2.cond
while.2.cond:
  br i1 0, label %while.2.loop, label %while.2.merge
while.2.loop:
  br i1 1, label %if.3.then, label %if.3.else
if.3.then:
  br label %if.3.merge
if.3.else:
  br label %if.3.merge
if.3.merge:
  br label %while.2.cond
while.2.merge:
  br label %if.0.merge
if.0.merge:
  ret i32 0
}


